mautic/mautic
# updated Hubspot integration to pull contacts and companies through a …

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | n
| New feature? | y
| Related user documentation PR URL | pending
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR adds a webhook that can be used in hubspot workflows, to push HS contacts into Mautic contacts, the same for companies

#### Steps to test this PR:
1. Apply PR
2. In the hubspot plugin go to the features tab and select the objects you wish to push from hubspot to mautic (contacts or companies)
3. Save and reopen the plugin. (working on a way to apply these changes without closing the modal)
4. match your Hubspot fields with Mautic fields. Save.
5. In your hubspot instance, create a new API key, this is the key that should be setup in your mautic plugin for hubspot.
6. Run the fetchleads command `php app/console mautic:integration:fetchleads --integration=Hubspot`
7. Test pushing a lead to integration - new changes. ~~Create a new property in your Hubspot contacts called Mautic Timeline (mautic_timeline as internal name).~~ _(not necessary, it does that automatically now)_
8. when pushing a lead to hubspot from Mautic check that the timeline link has been correctly written in the contact's info in hubspot.
